dtrace_provider.cc

```diff
@@ -92,6 +92,10 @@ namespace node {
     DTraceProbe *probe = Nan::ObjectWrap::Unwrap<DTraceProbe>(pd->ToObject());
     obj->Set(info[0]->ToString(), pd);
 
+    // reference the provider to avoid GC'ing it when only probes remain in scope.
+    Nan::ForceSet(pd, Nan::New<String>("__prov__").ToLocalChecked(), obj,
```

Comment:
This adds a new field to the probe object called `__prov__` and makes it a hidden, unmodifiable property. ForceSet is the Nan function to do this across different versions, since they changed the way to do it in one of the node versions I tested (v4, I think).